src/BenchmarkDotNet/Engines/IterationInfoExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Engines
+{
+    public static class IterationInfoExtensions
+    {
+        [PublicAPI] public static bool IsOverhead(this IIterationInfo iterationInfo) => iterationInfo.IterationMode == IterationMode.Overhead;
```

Comment:
when calling an interface method on a struct we introduce boxing

to avoid boxing, we need to make this method generic (however I am not 100% sure if an extension method can be generic):

```cs
public static bool IsOverhead<T>(this T iterationInfo) where T : IIterationInfo => iterationInfo.IterationMode == IterationMode.Overhead;
```

we try to avoid boxing at all cost in Engine, mostly to make sure that Engine does not allocate any managed memory which could spoil the results

more: https://adamsitnik.com/Value-Types-vs-Reference-Types/#boxing

Reply:
Yes Extension method can be generic. Before changes: 
![image](https://user-images.githubusercontent.com/17333903/51106168-47bf8d00-17eb-11e9-90b7-255d0d25b9e5.png)
After changes:
![image](https://user-images.githubusercontent.com/17333903/51106270-94a36380-17eb-11e9-94dd-387d47241f04.png)

Reply:
Thanks for this suggestion!